cmd/operator/internal/controller/kubearchiveconfig_controller.go

```diff
@@ -491,3 +587,48 @@ func (r *KubeArchiveConfigReconciler) removeFilters(ctx context.Context, kaconfi
 
 	return nil
 }
+
+func (r *KubeArchiveConfigReconciler) reconcileNamespace(ctx context.Context, kaconfig *kubearchivev1alpha1.KubeArchiveConfig) (*corev1.Namespace, error) {
+	log := log.FromContext(ctx)
+
+	log.Info("in reconcileNamespace")
+
+	ns := &corev1.Namespace{}
+	err := r.Get(ctx, types.NamespacedName{Name: kaconfig.Namespace}, ns)
+	if err == nil {
+		ns.ObjectMeta.Labels[ApiServerSourceLabelName] = ApiServerSourceLabelValue
+		err = r.Update(ctx, ns)
+		if err != nil {
+			log.Error(err, "Failed to update Namespace "+kaconfig.Namespace)
+			return ns, err
+		}
+	} else {
+		log.Error(err, "Failed to get Namespace "+kaconfig.Namespace)
+		return ns, err
+	}
```

Comment:
This is inconsistent with the error handling in the rest of the code. In the rest of the code, you check for a non-nil error return. Following that coding convention here would also allow you to reduce the depth of the conditional code:

```suggestion
	if err != nil {
		log.Error(err, "Failed to get Namespace "+kaconfig.Namespace)
		return ns, err
	}
	
	ns.ObjectMeta.Labels[ApiServerSourceLabelName] = ApiServerSourceLabelValue
	err = r.Update(ctx, ns)
	if err != nil {
		log.Error(err, "Failed to update Namespace "+kaconfig.Namespace)
		return ns, err
	}
```